src/Elastic.Transport/Responses/Special/StreamResponseBase.cs

```diff
@@ -0,0 +1,62 @@
+// Licensed to Elasticsearch B.V under one or more agreements.
+// Elasticsearch B.V licenses this file to you under the Apache 2.0 License.
+// See the LICENSE file in the project root for more information
+
+using System;
+using System.IO;
+
+namespace Elastic.Transport;
+
+/// <summary>
+/// A base class for implementing responses that access the raw response stream.
+/// </summary>
+public abstract class StreamResponseBase(Stream stream) : TransportResponse, IDisposable
```

Comment:
We should probably use a regular constructor here to validate that `stream` is not null.

LGTM otherwise 🙂